Services/UI/OJS.Services.Ui.Data/Implementations/ParticipantScoresDataService.cs

```diff
@@ -132,19 +134,32 @@ public async Task AddBySubmissionByUsernameAndIsOfficial(Submission submission,
                 IsOfficial = isOfficial,
             });
 
+            UpdateTotalScoreSnapshot(participant, 0, submission.Points, true);
+            this.participantsData.Update(participant);
+
             await this.SaveChanges();
         }
 
         public async Task UpdateBySubmissionAndPoints(
             ParticipantScore participantScore,
             int? submissionId,
-            int submissionPoints)
+            int submissionPoints,
+            Participant participant)
         {
             participantScore.SubmissionId = submissionId;
             participantScore.Points = submissionPoints;
 
+            var shouldUpdateTotalScoreDate = submissionId != null && submissionId != participantScore.SubmissionId;
```

Comment:
It would be very helpful if you add a comment here explaining why the TotalScoreSnapshotModifiedOn date should be updated only in this specific case.
The reason is that this date should stay the same after retest or normalization of points. The only case when it should be updated is when new submission is made and points are given for the first time to this submission. This will ensure proper arrangement in the leaderboard, as we want first person to receive max points to stay at firs place.. and so on.

Reply:
It's fixed.